it/controllers/pensions/unauthorisedPayments/UnauthorisedPensionSchemeTaxReferenceControllerISpec.scala

```diff
@@ -0,0 +1,368 @@
+    val expectedTitle = "Pension Scheme Tax Reference (PSTR)"
+    val expectedHeading = "Pension Scheme Tax Reference (PSTR)"
+    val expectedErrorTitle = s"Error: $expectedTitle"
+    val hintText = "For example, ’12345678RA’"
+  }
+
+  object CommonExpectedCY extends CommonExpectedResults {
+    val expectedCaption: Int => String = (taxYear: Int) => s"Taliadau heb awdurdod o bensiynau ar gyfer 6 Ebrill ${taxYear - 1} i 5 Ebrill $taxYear"
+    val expectedButtonText = "Continue"
+    val yesText = "Yes"
+    val noText = "No"
+    val expectedTitle = "Pension Scheme Tax Reference (PSTR)"
+    val expectedHeading = "Pension Scheme Tax Reference (PSTR)"
+    val expectedErrorTitle = s"Error: $expectedTitle"
+    val hintText = "For example, ’12345678RA’"
+  }
+
+  val inputName: String = "taxReferenceId"
+
+
+  val userScenarios: Seq[UserScenario[CommonExpectedResults, SpecificExpectedResults]] = Seq(
+    UserScenario(isWelsh = false, isAgent = false, CommonExpectedEN, Some(ExpectedIndividualEN)),
+    UserScenario(isWelsh = false, isAgent = true, CommonExpectedEN, Some(ExpectedAgentEN)),
+    UserScenario(isWelsh = true, isAgent = false, CommonExpectedCY, Some(ExpectedIndividualCY)),
+    UserScenario(isWelsh = true, isAgent = true, CommonExpectedCY, Some(ExpectedAgentCY))
+  )
+
+  ".show" should {
+    userScenarios.foreach { user =>
+      s"language is ${welshTest(user.isWelsh)} and request is from an ${agentTest(user.isAgent)}" should {
+
+        import Selectors._
+        import user.commonExpectedResults._
+
+        "render the 'PSTR' page with correct content and no pre-filling and no PSTR index" which {
+          implicit lazy val result: WSResponse = {
+            authoriseAgentOrIndividual(user.isAgent)
+            dropPensionsDB()
+            insertCyaData(anPensionsUserDataEmptyCya, aUserRequest)
+            urlGet(fullUrl(pensionSchemeTaxReferenceUrl(taxYearEOY)), user.isWelsh, follow = false,
+              headers = Seq(HeaderNames.COOKIE -> playSessionCookies(taxYearEOY, validTaxYearList)))
+          }
+
+          "has an OK status" in {
+            result.status shouldBe OK
+          }
+
+          implicit def document: () => Document = () => Jsoup.parse(result.body)
+
+          titleCheck(expectedTitle)
+          h1Check(expectedHeading)
+          captionCheck(expectedCaption(taxYearEOY))
+          textOnPageCheck(user.specificExpectedResults.get.expectedParagraph1, paragraphSelector(1))
+          textOnPageCheck(user.specificExpectedResults.get.expectedParagraph2, paragraphSelector(2))
+          textOnPageCheck(hintText, hintTextSelector)
+          inputFieldValueCheck(inputName, inputSelector, "")
+          buttonCheck(expectedButtonText, continueButtonSelector)
+          formPostLinkCheck(pensionSchemeTaxReferenceUrl(taxYearEOY), formSelector)
+          welshToggleCheck(user.isWelsh)
+        }
+
+        "render the 'PSTR' page with correct content with pre-filling and a PSTR index" which {
```

Comment:
How does this work ? As we don't have the ability to use a pstr index as its not been implemented yet